webui/react/src/components/UPlot/UPlotChart.tsx

```diff
@@ -219,18 +164,22 @@ const UPlotChart: React.FC<Props> = ({
         });
       }
     } else {
-      const isZoomed: boolean = Object.values(boundsRef.current).some((i) => i.isZoomed === true);
-      const propsProvideScales: boolean = Object.values(extendedOptions?.scales ?? {}).some(
-        (scale) => scale.min != null || scale.max != null,
-      );
-      const onlyOneXValue: boolean = Array.isArray(data) && data[0]?.length === 1;
-      const customScaleIsInEffect = isZoomed || (propsProvideScales && onlyOneXValue);
-      const resetScales = !customScaleIsInEffect;
-
       try {
         if (chartRef.current && isReady){
-          chartRef.current.setData(data as AlignedData, resetScales);
-          if (onlyOneXValue) chartRef.current.redraw(true, false);
+          const disp =
+          title === 'Throughput'
+            ? data.map((ser) => ser.slice(0, Math.floor(ser.length / 2)))
+            : data;
+          const min = new Date(Math.min(...disp[0]?.filter(isNumber)));
+          const max = new Date(Math.max(...disp[0]?.filter(isNumber)));
+          const minTime = min.toTimeString().slice(3, 9);
+          const maxTime = max.toTimeString().slice(3, 9);
+          // const minTime = `${min.getMinutes()}:${min.getSeconds()}`;
+          // const maxTime = `${max.getMinutes()}:${max.getSeconds()}`;
+          // console.log(data[0].length);
+          // console.log(title, minTime, maxTime);
+
+          chartRef.current.setData(disp as AlignedData, !zoomed);
```

Comment:
Depending on what comes up, it might be cleaner to have resetScales=false here and do set scales manually after getting syncMin and syncMax from the data, then dispatching the setScales manually. I think we can do the x axis manually and have the y axis automatic? 

We can also special case it with a prop so where for non-streaming data we are always updating scales